main.tf

```diff
@@ -149,13 +148,109 @@ resource "aws_launch_template" "this" {
+      accelerator_names         = try(instance_requirements.value.accelerator_names, [])
+
+      dynamic "accelerator_total_memory_mib" {
+        for_each = try([instance_requirements.value.accelerator_total_memory_mib], [])
+        content {
+          max = try(accelerator_total_memory_mib.value.max, null)
+          min = try(accelerator_total_memory_mib.value.min, null)
+        }
+      }
+
+      accelerator_types = try(instance_requirements.value.accelerator_types, [])
+      bare_metal        = try(instance_requirements.value.bare_metal, null)
+
+      dynamic "baseline_ebs_bandwidth_mbps" {
+        for_each = try([instance_requirements.value.baseline_ebs_bandwidth_mbps], [])
+        content {
+          max = try(baseline_ebs_bandwidth_mbps.value.max, null)
+          min = try(baseline_ebs_bandwidth_mbps.value.min, null)
+        }
+      }
+
+      burstable_performance   = try(instance_requirements.value.burstable_performance, null)
+      cpu_manufacturers       = try(instance_requirements.value.cpu_manufacturers, [])
+      excluded_instance_types = try(instance_requirements.value.excluded_instance_types, [])
+      instance_generations    = try(instance_requirements.value.instance_generations, [])
+      local_storage           = try(instance_requirements.value.local_storage, null)
+      local_storage_types     = try(instance_requirements.value.local_storage_types, [])
+
+      dynamic "memory_gib_per_vcpu" {
+        for_each = try([instance_requirements.value.memory_gib_per_vcpu], [])
+        content {
+          max = try(memory_gib_per_vcpu.value.max, null)
+          min = try(memory_gib_per_vcpu.value.min, null)
+        }
+      }
+
+      dynamic "memory_mib" {
+        for_each = try([instance_requirements.value.memory_mib], [])
+        content {
+          max = try(memory_mib.value.max, null)
+          min = try(memory_mib.value.min, null)
+        }
+      }
+
+      dynamic "network_interface_count" {
+        for_each = try([instance_requirements.value.network_interface_count], [])
+        content {
+          max = try(network_interface_count.value.max, null)
+          min = try(network_interface_count.value.min, null)
+        }
+      }
+
+      on_demand_max_price_percentage_over_lowest_price = try(instance_requirements.value.on_demand_max_price_percentage_over_lowest_price, null)
+      require_hibernate_support                        = try(instance_requirements.value.require_hibernate_support, null)
+      spot_max_price_percentage_over_lowest_price      = try(instance_requirements.value.spot_max_price_percentage_over_lowest_price, null)
+
+      dynamic "total_local_storage_gb" {
+        for_each = try([instance_requirements.value.total_local_storage_gb], [])
+        content {
+          max = try(total_local_storage_gb.value.max, null)
+          min = try(total_local_storage_gb.value.min, null)
+        }
+      }
+
+      dynamic "vcpu_count" {
```

Comment:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#vcpu_count - also should be `required` (according to the docs), same as `min`.

Reply:
Updated in https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/193/commits/ffec352d99cc67f8aef608c0c3b45677aa813582